src/rmarkdown.ts

```diff
@@ -43,67 +44,93 @@ export class RMarkdownCodeLensProvider implements CodeLensProvider {
   constructor() {
     this.decoration = window.createTextEditorDecorationType({
       isWholeLine: true,
-      backgroundColor: 'rgba(128, 128, 128, 0.1)',
+      backgroundColor: config().get('rmarkdown.chunkBackgroundColor'),
     });
   }
 
   public provideCodeLenses(document: TextDocument, token: CancellationToken): CodeLens[] | Thenable<CodeLens[]> {
     this.codeLenses = [];
-    const lines = document.getText().split(/\r?\n/);
-    let line = 0;
-    let chunkStartLine: number = undefined;
-    let chunkLanguage: string = undefined;
-    let chunkOptions: string = undefined;
+    const chunks = getChunks(document);
     const chunkRanges: Range[] = [];
-    const codeRanges: Range[] = [];
+    const rmdCodeLensCommands: string[] = config().get('rmarkdown.codeLensCommands');
 
-    while (line < lines.length) {
-      if (token.isCancellationRequested) {
-        break;
-      }
-      if (chunkStartLine === undefined) {
-        if (isChunkStartLine(lines[line])) {
-          chunkStartLine = line;
-          chunkLanguage = getChunkLanguage(lines[line]);
-          chunkOptions = getChunkOptions(lines[line]);
-        }
-      } else {
-        if (isChunkEndLine(lines[line])) {
-          const chunkRange = new Range(
-            new Position(chunkStartLine, 0),
-            new Position(line, lines[line].length)
-          );
-          const codeRange = new Range(
-            new Position(chunkStartLine + 1, 0),
-            new Position(line - 1, lines[line - 1].length)
-          );
-          chunkRanges.push(chunkRange);
-          if (chunkLanguage === 'r') {
-            this.codeLenses.push(new CodeLens(chunkRange, {
+    // Iterate through all code chunks for getting chunk information for both CodeLens and chunk background color (set by `editor.setDecorations`)
+    for (let i = 1 ; i <= chunks.length ; i++) {
+      const chunk = chunks.filter(e => e.id === i)[0];
+      const chunkRange = chunk.chunkRange;
+      const line = chunk.startLine;
+      chunkRanges.push(chunkRange);
+
+      // Enable/disable only CodeLens, without affecting chunk background color.
+      if (config().get<boolean>('rmarkdown.enableCodeLens')) {
```

Comment:
here can use && symbol

Reply:
Done in https://github.com/Ikuyadeu/vscode-R/pull/465/commits/5309e8c60dfa73992a1e6c36209e10df5b7a1e47